src/main/scala/commonist/task/ChangeDirectoryTask.scala

```diff
@@ -2,19 +2,34 @@ package commonist.task
 
 import java.io.File
 
+import scala.language.postfixOps
+import scala.xml._
+
+import scutil.base.implicits._
 import scutil.core.implicits._
+import scutil.io._
 import scutil.log._
 
 import commonist._
 import commonist.thumb._
 import commonist.ui._
 import commonist.ui.later._
+import commonist.util._
 
 /** change the directory displayed in the ImageListUI */
-final class ChangeDirectoryTask(mainWindow:MainWindow, imageListUI:ImageListUI, statusUI:StatusUI, thumbnails:Thumbnails, directory:File) extends Task {
+final class ChangeDirectoryTask(mainWindow:MainWindow, imageListUI:ImageListUI, statusUI:StatusUI, thumbnails:Thumbnails, directory:File, loader:Loader) extends Task {
 	private val imageListUILater	= new ImageListUILater(imageListUI)
 	private val statusUILater		= new StatusUILater(statusUI)
 
+	private def isXml(name:String):Boolean = {
+	  ".xml" == name.substring(name.lastIndexOf('.'))
+	}
+
+	private def getOaiPmhProps():Map[String,String] = {
+    val propsURL = (loader resourceURL "oaipmh.properties") getOrError "cannot load oaipmh.properties"
+    return PropertiesUtil loadURL (propsURL, None)
```

Comment:
you can (and should) leave out `return` here - a method always returns its last expression in scala